internal/command/binding_create.go

```diff
@@ -0,0 +1,326 @@
+			Namespace: namespace,
+			Name:      name,
+		},
+		Spec: v1alpha1.KameletBindingSpec{
+			Source: sourceEndpoint,
+			Sink:   sinkEndpoint,
+		},
+	}
+
+	existed := false
+	_, err = client.KameletBindings(namespace).Create(ctx, &binding, v1.CreateOptions{})
+	if err != nil && k8serrors.IsAlreadyExists(err) {
+		existed = true
+
+		existing, err := client.KameletBindings(namespace).Get(ctx, binding.Name, v1.GetOptions{})
+		if err != nil {
+			return nil, knerrors.GetError(err)
+		}
+		// Update the custom resource
+		binding.ResourceVersion = existing.ResourceVersion
+		_, err = client.KameletBindings(namespace).Update(ctx, &binding, v1.UpdateOptions{})
+		if err != nil {
+			return nil, knerrors.GetError(err)
+		}
+	}
+
+	if !existed {
+		fmt.Printf("kamelet binding \"%s\" created\n", name)
+	} else {
+		fmt.Printf("kamelet binding \"%s\" updated\n", name)
+	}
+
+	return &binding, nil
+}
+
+func nameFor(name, source string, sinkRef corev1.ObjectReference) string {
+	if name != "" {
+		return name
+	}
+
+	generated := fmt.Sprintf("%s-to-%s-%s", source, sinkRef.Kind, sinkRef.Name)
+
+	generated = filepath.Base(generated)
+	generated = strings.Split(generated, ".")[0]
+	generated = strings.ToLower(generated)
+	generated = disallowedChars.ReplaceAllString(generated, "")
+	generated = strings.TrimFunc(generated, isDisallowedStartEndChar)
+
+	return generated
+}
+
+func decodeSink(sink string) (corev1.ObjectReference, error) {
+	ref := corev1.ObjectReference{}
+
+	if sinkExpression.MatchString(sink) {
+		groupNames := sinkExpression.SubexpNames()
+		for _, match := range sinkExpression.FindAllStringSubmatch(sink, -1) {
+			for idx, text := range match {
+				groupName := groupNames[idx]
+				switch groupName {
+				case "apiVersion":
+					ref.APIVersion = text
+				case "namespace":
+					ref.Namespace = text
+				case "kind":
+					ref.Kind = text
+				case "name":
+					ref.Name = text
+				}
+			}
+		}
+
+		if sinkType, ok := sinkTypes[ref.Kind]; ok {
+			if sinkType.Kind != "" {
+				ref.Kind = sinkType.Kind
+			}
+			if ref.APIVersion == "" && sinkType.APIVersion != "" {
+				ref.APIVersion = sinkType.APIVersion
+			}
+		} else {
+			return ref, fmt.Errorf("unsupported sink type %q", ref.Kind)
+		}
+	} else {
+		return ref, fmt.Errorf("unsupported sink expression %q - please use format <kind>:<name>", sink)
+	}
+
+	return ref, nil
+}
+
+func verifyProperties(kamelet *v1alpha1.Kamelet, endpoint v1alpha1.Endpoint) error {
+	if kamelet.Spec.Definition != nil && len(kamelet.Spec.Definition.Required) > 0 {
+		pMap, err := endpoint.Properties.GetPropertyMap()
+		if err != nil {
+			return err
+		}
+		for _, reqProp := range kamelet.Spec.Definition.Required {
+			found := false
+			if endpoint.Properties != nil {
+				if _, contains := pMap[reqProp]; contains {
+					found = true
+				}
+			}
+			if !found {
+				return fmt.Errorf("binding is missing required property %q for Kamelet %q", reqProp, kamelet.Name)
+			}
+		}
+	}
+
```

Comment:
Would it make sense to also verify whether a given property is supported at all (not only required) ? E.g. to catch misspelling of property keys (can happen quite easily for the key=value way of providing properties)

Reply:
yes totally makes sense